src/main/java/nl/tudelft/contextproject/roomIO/EntityReader.java

```diff
@@ -0,0 +1,67 @@
+package nl.tudelft.contextproject.roomIO;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.List;
+
+import nl.tudelft.contextproject.model.Bomb;
+import nl.tudelft.contextproject.model.Door;
+import nl.tudelft.contextproject.model.Entity;
+import nl.tudelft.contextproject.model.Key;
+
+/**
+ * Utility class to read entities from a buffetredReader.
+ */
+public final class EntityReader {
+	private EntityReader() {}
+	
+	/**
+	 * Read the specified amount of entities and add them to the list of entities.
+	 * @param entities The list to add all loaded entities to.
+	 * @param entityCount The number of entities to load.
+	 * @param xOffset The horizontal offset used for all entities.
+	 * @param yOffset The vertical offset used for all entities.
+	 * @param br The bufferedReader to read from.
+	 * @throws IOException when reading from the reader goes wrong.
+	 */
+	public static void readEntities(List<Entity> entities, int entityCount, int xOffset, int yOffset, BufferedReader br) throws IOException {
+		for (int i = 0; i < entityCount; i++) {
+			String in = br.readLine();
+			if (in == null) throw new IllegalArgumentException("Empty line where some data was expected when loading entity[" + i + "].");
```

Comment:
That's because we want to allow inline ifs like these, where the first statement is a return, continue, break or throw (anything that directly exits the enclosing body)